packages/core/admin/admin/src/components/Theme.tsx

```diff
@@ -10,9 +11,29 @@ interface ThemeProps {
   children: React.ReactNode;
 }
 
+type NonSystemThemeName = Exclude<ThemeName, 'system'>;
+
 const Theme = ({ children }: ThemeProps) => {
   const { currentTheme, themes } = useThemeToggle();
   const { locale } = useIntl();
+  const [systemTheme, setSystemTheme] = React.useState<NonSystemThemeName>();
+
+  // Listen to changes in the system theme
+  React.useEffect(() => {
+    const themeWatcher = window.matchMedia('(prefers-color-scheme: dark)');
+    setSystemTheme(themeWatcher.matches ? 'dark' : 'light');
+
+    themeWatcher.addEventListener('change', (event) => {
+      setSystemTheme(event.matches ? 'dark' : 'light');
+    });
+
+    // Cleanup on unmount
+    return () => {
+      themeWatcher.removeEventListener('change', () => {});
+    };
+  }, []);
```

Comment:
```suggestion
  React.useEffect(() => {
		const listener = (event) => {
			setSystemTheme(event.matches ? 'dark' : 'light');
		}
    const themeWatcher = window.matchMedia('(prefers-color-scheme: dark)');
    setSystemTheme(themeWatcher.matches ? 'dark' : 'light');

    themeWatcher.addEventListener('change', listener);
    // Cleanup on unmount
    return () => {
      themeWatcher.removeEventListener('change', listener);
    };
  }, []);
```

I think the way you're doing doesn't remove the handler? could be wrong.

I also wonder if we should only update the system theme if the theme toggle is set to system to avoid unnecessary set state calls?

Reply:
Yes reading my code again it definitely doesn't remove it